File: CS-Examples/13_MailMerge/MailMergeSwitches.cs
using System;
using System.Windows.Forms;
using System.Text;
using Spire.Doc;
using Spire.Doc.Documents;
using System.Drawing;
using Spire.Doc.Formatting;
using System.IO;
using Spire.Doc.Fields;
using System.Data;
using System.Data.OleDb;
namespace MailMergeSwitches
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            string input = @"..\..\..\..\..\..\Data\MailMergeSwitches.docx";

			//Create a Word document
			Document doc = new Document();

			//Load a mail merge template file
			doc.LoadFromFile(input);

			//Define the field names for the mail merge
			string[] fieldName = new string[] { "XX_Name" };

			//Define the field values for the mail merge
			string[] fieldValue = new string[] { "Jason Tang" };

			//Execute the mail merge using the field names and values
			doc.MailMerge.Execute(fieldName, fieldValue);

			//Save to file
			string result = "MailMergeSwitches_out.docx";
			doc.SaveToFile(result, FileFormat.Docx);

			// Dispose the document object
			doc.Dispose();
            WordViewer(result);
        }      
        private void WordViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB-Examples/13_MailMerge/MailMergeSwitches.vb
Imports System.Text
Imports Spire.Doc
Imports Spire.Doc.Documents
Imports Spire.Doc.Formatting
Imports System.IO
Imports Spire.Doc.Fields
Imports System.Data.OleDb
Namespace MailMergeSwitches
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			Dim input As String = "..\..\..\..\..\..\Data\MailMergeSwitches.docx"

			'Create a Word document
			Dim doc As New Document()

			'Load a mail merge template file
			doc.LoadFromFile(input)

			'Define the field names for the mail merge
			Dim fieldName() As String = {"XX_Name"}

			'Define the field values for the mail merge
			Dim fieldValue() As String = {"Jason Tang"}

			'Execute the mail merge using the field names and values
			doc.MailMerge.Execute(fieldName, fieldValue)
			
			'Save to file
			Dim result As String = "MailMergeSwitches_out.docx"
			doc.SaveToFile(result, FileFormat.Docx)

			'Dispose the document
			doc.Dispose()
			
			WordViewer(result)
		End Sub
		Private Sub WordViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace
